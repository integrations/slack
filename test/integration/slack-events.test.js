const request = require('supertest');

const { probot } = require('.');

describe('Slack events', () => {
  describe('POST /slack/events.url_verification', () => {
    test('responds with challenge if token matches', async () => {
      await request(probot.server).post('/slack/events')
        .send({
          token: process.env.SLACK_VERIFICATION_TOKEN,
          challenge: 'abc-123',
          type: 'url_verification',
        })
        .expect(200, 'abc-123');
    });

    test('responds with 400 if token is invalid', async () => {
      await request(probot.server).post('/slack/events')
        .send({
          token: 'wrong-togken',
          challenge: 'abc-123',
          type: 'url_verification',
        })
        .expect(400);
    });

    test('responds with 400 when POSTed to directly', async () => {
      await request(probot.server).post('/slack/events.config_migration')
        .send({})
        .expect(400)
        .expect('Invalid verificaton token');
    });
  });
});
