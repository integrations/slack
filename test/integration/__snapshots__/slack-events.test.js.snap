// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack events POST /slack/events.url_verification responds with 400 if token is invalid 1`] = `Set {}`;

exports[`Slack events POST /slack/events.url_verification responds with 400 if token is invalid 2`] = `Map {}`;

exports[`Slack events POST /slack/events.url_verification responds with challenge if token matches 1`] = `Set {}`;

exports[`Slack events POST /slack/events.url_verification responds with challenge if token matches 2`] = `Map {}`;
